### PR TITLE
eth-feehistory docs

### PR DESCRIPTION
**Pull request by luu-alex**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../CONTRIBUTIONS.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: docs/web3-eth.rst [hunk 1]
```diff
@@ -716,6 +716,43 @@ Example
...
+Transaction fee history
+Returns base fee per gas and transaction effective priority fee per gas history for the requested block range if available. 
+The range between headBlock-4 and headBlock is guaranteed to be available while retrieving data from the pending block and older
+history are optional to support. For pre-EIP-1559 blocks the gas prices are returned as rewards and zeroes are returned for the base fee per gas.
+
+----------
+Parameters
+----------
+
+1. ``String`` - Number of blocks in the requested range. Between 1 and 1024 blocks can be requested in a single query. Less than requested may be returned if not all blocks are available.
```
Comment (Spacesai1or) [inline docs/web3-eth.rst]:
```suggestion
1. ``String|Number|BN|BigNumber`` - Number of blocks in the requested range. Between 1 and 1024 blocks can be requested in a single query. Less than requested may be returned if not all blocks are available.
```

File: docs/web3-eth.rst [hunk 2]
```diff
@@ -716,6 +716,43 @@ Example
...
+2. ``String`` - Highest number block of the requested range.
```
Comment (Spacesai1or) [inline docs/web3-eth.rst]:
```suggestion
2. ``String|Number|BN|BigNumber`` - Highest number block of the requested range.
```

File: docs/web3-eth.rst (hunk 1 above)
Comment (corymsmith) [reply docs/web3-eth.rst]: Shouldn't `blockCount` just be a `number`? The method takes an int, https://github.com/ethereum/go-ethereum/blob/c49e065fea78a5d3759f7853a608494913e5824e/internal/ethapi/api.go#L90

I guess it depends on if an inputFormatter is applied to force it to format it to a number?

Comment (GregTheGreek) [reply docs/web3-eth.rst]: Unfortunately its a design decision we need to stick with based on the history of web3.js, the 4.x branch improves on this a bit.

For reference check out [eth_gettransactionfromblock](https://web3js.readthedocs.io/en/v1.4.0/web3-eth.html#gettransactionfromblock)

File: docs/web3-eth.rst [hunk 3]
```diff
@@ -716,6 +716,43 @@ Example
...
+3. ``Array of numbers`` - (optional) A monotonically increasing list of percentile values to sample from each block's effective priority fees per gas in ascending order, weighted by gas used.
```
Comment (GregTheGreek) [inline docs/web3-eth.rst]: can we maybe include an example of a `monotonically` list

File: docs/web3-eth.rst [hunk 4]
```diff
@@ -716,6 +716,43 @@ Example
...
+The range between headBlock-4 and headBlock is guaranteed to be available while retrieving data from the pending block and older
+history are optional to support. For pre-EIP-1559 blocks the gas prices are returned as rewards and zeroes are returned for the base fee per gas.
+
+----------
+Parameters
+----------
+
+1. ``String|Number|BN|BigNumber`` - Number of blocks in the requested range. Between 1 and 1024 blocks can be requested in a single query. Less than requested may be returned if not all blocks are available.
+2. ``String|Number|BN|BigNumber`` - Highest number block of the requested range.
+3. ``Array of numbers`` - (optional) A monotonically increasing list of percentile values to sample from each block's effective priority fees per gas in ascending order, weighted by gas used. Example: `["0", "25", "50", "75", "100"]` or `["0", "0.5", "1", "1.5", "3", "80"]`
```
Comment (Spacesai1or) [inline docs/web3-eth.rst]:
```suggestion
3. ``Array of numbers`` - A monotonically increasing list of percentile values to sample from each block's effective priority fees per gas in ascending order, weighted by gas used. Example: `["0", "25", "50", "75", "100"]` or `["0", "0.5", "1", "1.5", "3", "80"]`
```

File: docs/web3-eth.rst [hunk 5]
```diff
@@ -716,6 +716,43 @@ Example
...
+4. ``Function`` - (optional) Optional callback, returns an error object as first parameter and the result as second.
+
+-------
+Returns
+-------
+
+``Promise`` returns ``Object`` - Fee history for the returned block range. The object:
+
+- ``String`` oldestBlock  - Lowest number block of the returned range.
```
Comment (Spacesai1or) [inline docs/web3-eth.rst]:
```suggestion
- ``Number`` oldestBlock  - Lowest number block of the returned range.
```

File: docs/web3-eth.rst [hunk 6]
```diff
@@ -716,6 +716,43 @@ Example
...
+- ``Array of strings`` baseFeePerGas  - An array of block base fees per gas. This includes the next block after the newest of the returned range, because this value can be derived from the newest block. Zeroes are returned for pre-EIP-1559 blocks.
+- ``Array of numbers`` gasUsedRatio  - An array of block gas used ratios. These are calculated as the ratio of gasUsed and gasLimit.
+- ``Array of strings`` reward  - An array of effective priority fee per gas data points from a single block. All zeroes are returned if the block is empty.
```
Comment (Spacesai1or) [inline docs/web3-eth.rst]:
```suggestion
- ``Array of string arrays`` reward  - An array of effective priority fee per gas data points from a single block. All zeroes are returned if the block is empty.
```

**Review by Spacesai1or — APPROVED** [review]

**Pull request merged**